### PR TITLE
[FORMS-11581] Enabled Draft functionality for V2

### PR DESCRIPTION
**Pull request by girotraapankaj**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes and the overall coverage did not decrease.
- [x] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/site/js/formcontainerview.js [hunk 1]
```diff
@@ -57,6 +57,21 @@
                     FormView.customFunctions.defaultSubmitErrorHandler(defaultSubmissionError, globals);
                 }
             }, "submitError");
+            this._model.subscribe((action) => {
```
Comment (arunattri26) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/site/js/formcontainerview.js:60]: Have we checked that we don't need to override hardcoded save handlers like we have done for submit? Or we will provide it later if customer ask?

**Review by arunattri26 — APPROVED** [review]

**Comment by rismehta** [comment]
Please don't merge until my review is complete

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/site/js/formcontainerview.js [hunk 2]
```diff
@@ -57,6 +57,21 @@
...
+                let state = action.target.getState();
+                // execute the handler only if there are no rules configured on submitSuccess event.
+                if (!state.events['custom:saveSuccess'] || state.events['custom:saveSuccess'].length === 0) {
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/site/js/formcontainerview.js:63]: This looks incorrect to me, you have hardcoded the view here and the same code would not work for headless too. Can you change the code and implement it the same way as done in submit ? Additionallly, you also need to fix this, https://github.com/adobe/react-starter-kit-aem-headless-forms/blob/main/src/Form.tsx#L50-L83

File: ui.tests/test-module/specs/ruleeditor/ruleEditor.runtime.spec.js [hunk 3]
```diff
@@ -142,3 +142,42 @@ describe("Rule editor submission handler runtime", () => {
...
+    const saveRunTime = "content/forms/af/core-components-it/samples/ruleeditor/save/saveruntime.html"
+
+    it("should save formData on button click", () => {
+        if (toggle_array.includes("FT_FORMS-11581")) {
+
+            const saveApiResponse = {
+                'draftId': 'ABC'
+            };
+            // Rule when button is clicked then save call should trigger
+            cy.intercept('POST' , '**/adobe/forms/af/save/*', saveApiResponse).as('afSave');
```
Comment (rismehta) [inline ui.tests/test-module/specs/ruleeditor/ruleEditor.runtime.spec.js:167]: Why do we need mock for e2e test case, can you please test the actual behavior ?

**Review by rismehta — CHANGES_REQUESTED** [review]
check comments

File: ui.tests/test-module/specs/ruleeditor/ruleEditor.runtime.spec.js (hunk 3 above)
Comment (girotraapankaj) [reply ui.tests/test-module/specs/ruleeditor/ruleEditor.runtime.spec.js:167]: OOTB draft gets stored in Azure, and if we provide azure configuration for test cases, then that will get polluted over time. So I discussed with Rahul and decided to mock the API.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/site/js/formcontainerview.js (hunk 2 above)
Comment (girotraapankaj) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/site/js/formcontainerview.js:63]: As of now, we are hardcoding save success and failure event and not asking from customer. 
And, here we are checking if these events are registered through rule (save success and save error), then we are not executing default functionality. Submit is also implemented the same way here.

https://github.com/adobe/react-starter-kit-aem-headless-forms/blob/main/src/Form.tsx#L50-L83 -> will look into this.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/site/js/formcontainerview.js (hunk 1 above)
Comment (girotraapankaj) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/site/js/formcontainerview.js:60]: As of now, we dont have any customer for this so we can do once some one asks for this.

**Pull request merged**